SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2Producer.cc

```diff
@@ -0,0 +1,185 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/Provenance/interface/ProductID.h"
+#include "DataFormats/Provenance/interface/ParameterSetID.h"
+#include "DataFormats/Provenance/interface/Provenance.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CondFormats/EcalObjects/interface/EcalLiteDTUPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalLiteDTUPedestalsRcd.h"
+
+//#include "CondFormats/EcalObjects/interface/EcalTPGTowerStatus.h"
+//#include "CondFormats/DataRecord/interface/EcalTPGStripStatusRcd.h"
+//#include "CondFormats/EcalObjects/interface/EcalTPGStripStatus.h"
+
+#include "EcalEBTrigPrimPhase2Producer.h"
```

Comment:
The class definition should be in the .cc file and not in a separate header.

Reply:
Again ... why ? If it's a rule it's escaping me

Reply:
There is no need for any other body of code to see the code in the header. Therefore we've moved to putting the class declaration in the same file as the class definition. This makes code maintenance easier and avoids someone trying to include that header (which wouldn't work right anyway as one can't link to the shared object holding the plugin).